### ISSUE TITLE
[env python] failed on MacOS 12.3, as python2 removed

### ISSUE THREAD

**Issue by honwen** [bazelbuild/rules_python#660]
# 🐞 bug report

### Affected Rule

 The issue is caused by the rule: 

### Is this a regression?

 Yes, the previous version in which this bug was not present was: ....
MacOS 12.2.1

### Description

 A clear and concise description of the problem...

## 🔬 Minimal Reproduction

OS: MacOS 12.3
Bazel: 5.0.0

```shell
bazel5 run //examples:py_import_example
```

## 🔥 Exception or Error

<pre><code>

INFO: Analyzed 2 targets (24 packages loaded, 318 targets configured).
INFO: Found 1 target and 1 test target...
[0 / 5] [Prepa] BazelWorkspaceStatusAction stable-status.txt ... (4 actions, 0 running)
FAIL: //:py_import_test (see /private/var/tmp/***/execroot/py_import/bazel-out/darwin-fastbuild/testlogs/py_import_test/test.log)
INFO: From Testing //:py_import_test:
==================== Test output for //:py_import_test:
env: python: No such file or directory
================================================================================
INFO: Elapsed time: 93.025s, Critical Path: 2.22s
INFO: 6 processes: 4 internal, 2 darwin-sandbox.
INFO: Build completed, 1 test FAILED, 6 total actions
//:py_import_test                                                        FAILED in 0.8s
  /private/var/tmp/***/execroot/py_import/bazel-out/darwin-fastbuild/testlogs/py_import_test/test.log

Executed 1 out of 1 test: 1 fails locally.
INFO: Build completed, 1 test FAILED, 6 total actions
copying workspace under test examples/py_import to /private/var/tmp/***/execroot/rules_python/_tmp/12d36418595ed21d10a139c999170d5f/tmp4adyy1pw/wksp
</code></pre>

`testlogs/py_import_test/test.log`:
```
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //:py_import_test
-----------------------------------------------------------------------------
env: python: No such file or directory
```

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
MacOS 12.3
M1 pro
  </code>
</pre>

**Output of `bazel version`:**

<pre>
  <code>
5.0.0
  </code>
</pre>

**Rules_python version:**

<pre>
  <code>
commit/e18e1a3bbd372fd0351277794155b5b9045fbe37
  </code>
</pre>

**Anything else relevant?**

**Comment by honwen** [comment]
> Thanks for your issue report. Unfortunately, by default `rules_python` will lookup `python` on the system in its stub script.
> 
> The workarounds are:
> 
> * install `python` onto the system or alias an existing `python3`.
> * Replace `python` in the stub script, similar to [Add ability for Bazel >=5.0.0 users to use downloaded py interpreter in py_binary stub script #699](https://github.com/bazelbuild/rules_python/pull/699).

Unfortunately, rules_apple does not seem to export the PATH set by the user (cf. [run.use_default_shell_env](https://bazel.build/rules/lib/actions#run.use_default_shell_env))

so this can not work
> install python onto the system or alias an existing python3.

**Comment by groodt** [comment]
Duplicate of: https://github.com/bazelbuild/rules_python/issues/691

**Comment by honwen** [comment]
A solution showcase

```python
py_runtime(
    name = "python3_runtime",
    files = ["@python_interpreter//:files"],
    interpreter = "@python_interpreter//:interpreter",
    python_version = "PY3",
    # https://github.com/bazelbuild/bazel/pull/11434
    # https://github.com/bazelbuild/rules_python/issues/660
    stub_shebang = select({
        # fix macos>=12.3 without python
        "//third_party/prebuild:macos": "#!/usr/bin/env python3",
        "//conditions:default": "",
    }),
    visibility = ["//visibility:public"],
)
```
